server/src/main/java/org/elasticsearch/search/aggregations/pipeline/InternalPercentilesBucket.java

```diff
@@ -69,6 +71,7 @@ public InternalPercentilesBucket(StreamInput in) throws IOException {
         format = in.readNamedWriteable(DocValueFormat.class);
         percentiles = in.readDoubleArray();
         percents = in.readDoubleArray();
+        keyed = in.readBoolean();
```

Comment:
We'll need to enclose all the new `keyed` values in version checks for the serialization code.  Otherwise, in mixed clusters, a node without this feature will be serialized/deserialized incorrectly because everyone is assuming offsets are in a different position :)

The format is generally something like:

```
if (in.getVersion().onOrAfter(Version.V_7_0_0)) {
  keyed = in.readBoolean();
}
```

This pattern applies to basically all the StreamInput/StreamOutput methods